### PR TITLE
Fix status command

### LINKED ISSUE #208: Copying dbase.sqlite from Registry causes ddb status to report wrong information
I think the logic in `ddb pull` and `ddb clone` where the remote `dbase.sqlite` is simply copied to update the DroneDB index needs some more thought; specifically, if I clone a repository and issue:

`ddb status`

Datasets that contain subfolders report those subfolders as been modified (M).

```
[piero:/data/test] $ ddb clone http://localhost:5000/admin/test33
Cloning dataset 'admin/test33' from registry 'http://localhost:5000' to folder '/data/test/test33'
Dataset downloaded (0 B)		
Extracting to destination folder (this could take a while)
Done
[piero:/data/test] $ cd test33
[piero:/data/test/test33] $ ddb status
M	a
M	images
[piero:/data/test/test33] $ 
```

`a` and `images` are folders.

### PR DESCRIPTION
**Pull request by HeDo88TH**
Closes #208 
Simple refactor and changed behaviour of the function `checkUpdate`: it now skips mtime check for folders.

### CODE REVIEW

File: src/cmd/status.cpp [hunk 1]
```diff
@@ -2,8 +2,10 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-#include <iostream>
 #include "status.h"
+
+#include <iostream>
```
Comment (pierotofy) [inline src/cmd/status.cpp:7]: Can we please disable auto-code formatting? :)

**Comment by pierotofy** [comment]
Ok I can't discern what was done by the auto-formatter and what was actually changed; please re-open another PR this time disabling the auto-formatting changes.

File: src/cmd/status.cpp (hunk 1 above)
Comment (HeDo88TH) [reply src/cmd/status.cpp:7]: The formatter just replaced tabs with spaces and swapped a couple of includes.

**Comment by HeDo88TH** [comment]
> Ok I can't discern what was done by the auto-formatter and what was actually changed; please re-open another PR this time disabling the auto-formatting changes.

The auto formatter just replaced tabs with spaces and sorted the includes.

Here are the actual code changes:

- Added `NotModified` enum value to `FileStatus`
- Moved `checkUpdate` function declaration from `dbops.h` to `status.h`
- `checkUpdate` function now returns a `FileStatus` and checks internally if the file/folder exists
- `checkUpdate` function skips `mtime` check if it's a folder

**Comment by pierotofy** [comment]
> please re-open another PR this time disabling the auto-formatting changes.

**Comment by HeDo88TH** [comment]
@pierotofy reverted all auto format changes. 
Soon or later we should clean up the codebase because many files have different coding styles, spacing and indentation conventions. I will open a PR to do this.

**Review by pierotofy — CHANGES_REQUESTED** [review]
Thanks. :+1: 

This still needs some work.

When you change a function's signature, search the codebase where it's used; you will find that it's being called in:

 - `dbops.cpp:287`, `dbops.cpp:486` and in both places, it breaks.

```
[piero:/data/test/test] $ ddb sync
U	a
[piero:/data/test/test] $ ddb sync
U	a
[piero:/data/test/test] $ ddb sync
U	a
[piero:/data/test/test] $ ddb sync
U	a
[piero:/data/test/test] $
```

**Comment by pierotofy** [comment]
Attaching below the checklist for PRs:

 - [ ] Compiles on Windows and Linux
 - [ ] Manual tests on both Windows and Linux to confirm the functionality/command works
 - [ ] Manual tests on commands/functionality that might have been affected by these changes.
 - [ ] There are no styling changes to unrelated code.

**Comment by pierotofy** [comment]
Thanks! :+1:  Looks good.

**Pull request merged**